src/HttpServer.jl

```diff
@@ -89,15 +89,29 @@ handler.events["foo"] = (bar) "Hello \$bar"
 HttpServer.event(server, "foo", "Julia")
 ```
 """
+
 immutable HttpHandler
     handle::Function
     sock::Base.TCPServer
     events::Dict
 
-    HttpHandler(handle::Function, sock::Base.TCPServer) = new(handle, sock, defaultevents)
-    HttpHandler(handle::Function) = new(handle, Base.TCPServer(), defaultevents)
+    function HttpHandler(handle::Function, sock::Base.TCPServer)
+	new(handle, sock, defaultevents)
+    end
 end
-handle(handler::HttpHandler, req::Request, res::Response) = handler.handle(req, res)
+HttpHandler(handle::Function) = HttpHandler(handle, Base.TCPServer())
+
+function handle(handler::HttpHandler, req::Request, res::Response)
+    # Uses try/catch because method_exists doesn't work for anonymous functions...i.e., do syntax.
+    try
+        @which handler.handle(Request())    # Check if signature of handle is (:Request) 
+	handler.handle(req)
+    catch err
+        Base.warn_once("The use of handler(Request, Response) is deprecated. Please use handler(Request) instead.")
```

Comment:
This will also catch errors thrown by the handler, even if the one argument handler does exist